app/Http/Controllers/UserController.php

```diff
@@ -211,26 +212,58 @@ public static function getProfilePage($username): ?array {
     }
 
     /**
-     * Add $userToFollow to $user's Follower
-     * @param User $user
-     * @param User $userToFollow The user of the person who is followed
+     * Add $userToFollow to $user's Followings
+     * @param User $user The user who initiated the follow
+     * @param User $userToFollow The user who is followed
+     * @param bool $isApprovedRequest Differentiates between who to be notified
      * @return bool
+     * @throws AlreadyFollowingException
      */
-    public static function createFollow(User $user, User $userToFollow): bool {
+    public static function createFollow(User $user, User $userToFollow, bool $isApprovedRequest = false): bool {
+        //disallow re-following, if you already follow them
+        //Also disallow following, if user is a private profile
         if (self::isFollowing($user, $userToFollow)) {
-            return false;
+            throw new AlreadyFollowingException($user, $userToFollow);
+        }
+        // Request follow if user is a private profile
+        if ($userToFollow->private_profile && !$isApprovedRequest) {
+            return self::requestFollow($user, $userToFollow);
         }
 
         $follow = Follow::create([
                                      'user_id'   => $user->id,
                                      'follow_id' => $userToFollow->id
                                  ]);
-
-        $userToFollow->notify(new UserFollowed($follow));
+        if (!$isApprovedRequest) {
+            $userToFollow->notify(new UserFollowed($follow));
+        } else {
+            $user->notify(new FollowRequestApproved($follow));
+        }
         $user->load('follows');
         return self::isFollowing($user, $userToFollow);
     }
 
+    /**
+     * Add $userToFollow to $user's FollowerRequests
+     * @param User $user
+     * @param User $userToFollow The user of the person who is followed
+     * @return bool
+     * @throws AlreadyFollowingException
+     */
+    public static function requestFollow(User $user, User $userToFollow): bool {
+        if ($userToFollow->followRequests->contains('user_id', $user->id)) {
+            throw new AlreadyFollowingException($user, $userToFollow);
```

Comment:
You throw an AlreadyFollowingException when one wants to request and has already requested? Wouldn't it make more sense to throw an AlreadyFollowRequestedException (or similar)? Otherwise the messages are confusing for the user.

Reply:
I'm not sure if this really needs a new exception type since the handling and message are done elsewhere and the frontend knows if you issue a request or a create. I thought, another Exception containing basically the same logic except for the name would be a bit useless.

Reply:
Okay!